src/mixxx.cpp

```diff
@@ -82,338 +82,52 @@
-
-    QString settingsPath = args.getSettingsPath();
-#ifdef __APPLE__
-    if (!args.getSettingsPathSet()) {
-        settingsPath = Sandbox::migrateOldSettings();
-    }
-#endif
-
-    mixxx::Logging::initialize(
-            settingsPath,
-            args.getLogLevel(),
-            args.getLogFlushLevel(),
-            args.getDebugAssertBreak());
-
-    VERIFY_OR_DEBUG_ASSERT(SoundSourceProxy::registerProviders()) {
-        qCritical() << "Failed to register any SoundSource providers";
-        return;
-    }
-
-    Version::logBuildDetails();
-
-    // Only record stats in developer mode.
-    if (m_cmdLineArgs.getDeveloper()) {
-        StatsManager::createInstance();
-    }
-
-    m_pSettingsManager = std::make_unique<SettingsManager>(args.getSettingsPath());
-
-    initializeKeyboard();
-    installEventFilter(m_pKeyboard);
-
-    // Menubar depends on translations.
-    mixxx::Translations::initializeTranslations(
-        m_pSettingsManager->settings(), pApp, args.getLocale());
-
+    DEBUG_ASSERT(pApp);
+    DEBUG_ASSERT(pCoreServices);
+    m_pCoreServices->initializeSettings();
+    m_pCoreServices->initializeKeyboard();
+    // These depend on the settings
     createMenuBar();
-
     initializeWindow();
 
-    // First load launch image to show a the user a quick responds
-    m_pSkinLoader = new SkinLoader(m_pSettingsManager->settings());
+    // Show launch image immediately so the user knows Mixxx is starting
+    m_pSkinLoader = std::make_unique<SkinLoader>(m_pCoreServices->getSettings());
     m_pLaunchImage = m_pSkinLoader->loadLaunchImage(this);
     m_pCentralWidget = (QWidget*)m_pLaunchImage;
     setCentralWidget(m_pCentralWidget);
 
     show();
     pApp->processEvents();
 
-    initialize(pApp, args);
-}
-
-MixxxMainWindow::~MixxxMainWindow() {
-    finalize();
-    // SkinLoader depends on Config;
-    delete m_pSkinLoader;
-}
-
-void MixxxMainWindow::initialize(QApplication* pApp, const CmdlineArgs& args) {
-    ScopedTimer t("MixxxMainWindow::initialize");
-
-#if defined(Q_OS_LINUX)
-    // XESetWireToError will segfault if running as a Wayland client
-    if (pApp->platformName() == QLatin1String("xcb")) {
-        for (auto i = 0; i < NUM_HANDLERS; ++i) {
-            XESetWireToError(QX11Info::display(), i, &__xErrorHandler);
-        }
-    }
-#endif
-
-    UserSettingsPointer pConfig = m_pSettingsManager->settings();
-
-    Sandbox::initialize(QDir(pConfig->getSettingsPath()).filePath("sandbox.cfg"));
-
-    QString resourcePath = pConfig->getResourcePath();
-
-    FontUtils::initializeFonts(resourcePath); // takes a long time
-
-    launchProgress(2);
-
-    // Set the visibility of tooltips, default "1" = ON
-    m_toolTipsCfg = static_cast<mixxx::TooltipsPreference>(
-        pConfig->getValue(ConfigKey("[Controls]", "Tooltips"),
-                static_cast<int>(mixxx::TooltipsPreference::TOOLTIPS_ON)));
-
-    m_pTouchShift = new ControlPushButton(ConfigKey("[Controls]", "touch_shift"));
```

Comment:
This PR was merged prematurely. I now get debug assertions on `main`:
```
warning [Main] ControlDoublePrivate::getControl returning NULL for ( "[Controls]" , "touch_shift" )
critical [Main] DEBUG ASSERT: "flags.testFlag(ControlFlag::NoAssertIfMissing)" in function static QSharedPointer<ControlDoublePrivate> ControlDoublePrivate::getControl(const ConfigKey&, ControlFlags, ControlObject*, bool, bool, bool, double) at /home/jan/Projects/mixxx/src/control/control.cpp:151
critical [Main] DEBUG ASSERT: "m_pControl || flags.testFlag(ControlFlag::NoAssertIfMissing)" in function void ControlProxy::initialize(ControlFlags) at /home/jan/Projects/mixxx/src/control/controlproxy.cpp:37
critical [Main] DEBUG ASSERT: "valid() || flags.testFlag(ControlFlag::NoAssertIfMissing)" in function void ControlProxy::initialize(ControlFlags) at /home/jan/Projects/mixxx/src/control/controlproxy.cpp:38
```

Reply:
fixed in #3453